### PR TITLE
global: Normalize isIpv4/6 to isIPv4/6

### PR DESCRIPTION
**Pull request by cmouse**
### Short description
This changes the Netmask class to use isIPv6 instead of isIpv6.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [ ] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/lua-base4.cc [hunk 1]
```diff
@@ -166,7 +166,7 @@ void BaseLua4::prepareContext() {
   d_lw->registerFunction<ComboAddress(Netmask::*)()>("getNetwork", [](const Netmask& nm) { return nm.getNetwork(); } ); // const reference makes this necessary
   d_lw->registerFunction<ComboAddress(Netmask::*)()>("getMaskedNetwork", [](const Netmask& nm) { return nm.getMaskedNetwork(); } );
   d_lw->registerFunction("isIpv4", &Netmask::isIpv4);
-  d_lw->registerFunction("isIpv6", &Netmask::isIpv6);
```
Comment (rgacogne) [inline pdns/lua-base4.cc]: The existing version is documented in docs/lua-records/reference/netmask.rst and pdns/recursordist/docs/lua-scripting/netmask.rst so we probably should not remove it right away. We could perhaps add the new version and deprecate the existing one?

**Comment by cmouse** [comment]
Added Ipv4 => IPv4 normalization too as it was missing.

File: docs/lua-records/reference/netmask.rst [hunk 2]
```diff
@@ -42,10 +42,22 @@ The :class:`Netmask` class represents an IP netmask.
       Returns a :class:`ComboAddress` representing the network (truncating according to the mask).
 
   .. method:: Netmask:isIpv4() -> bool
+  .. deprecated:: v4.x.x
```
Comment (rgacogne) [inline docs/lua-records/reference/netmask.rst]: I believe we need an empty line before `deprecated` and `versionadded`, see #8437. Same thing in pdns/recursordist/docs/lua-scripting/netmask.rst, unfortunately :'(

Comment (cmouse) [reply docs/lua-records/reference/netmask.rst]: fixed.

**Comment by Habbie** [comment]
Can you:
* update the doc changes to say deprecated/added in 4.3.0
* add the news to the auth and recursor upgrade notes?

Does this not affect dnsdist?

**Comment by cmouse** [comment]
> Can you:
> 
>     * update the doc changes to say deprecated/added in 4.3.0
> 
>     * add the news to the auth and recursor upgrade notes?
> 

Sure

> 
> Does this not affect dnsdist?

Does not seem to.

**Comment by cmouse** [comment]
requested changes made.

File: docs/upgrading.rst [hunk 3]
```diff
@@ -8,6 +8,11 @@ Please upgrade to the PowerDNS Authoritative Server 4.0.0 from 3.4.2+.
 See the `3.X <https://doc.powerdns.com/3/authoritative/upgrading/>`__
 upgrade notes if your version is older than 3.4.2.
 
+4.2.x to 4.3.0
+--------------
+
+- Netmask class methods ``isIpv4`` and ``isIpv6`` has been deprecated in Lua, use :func:`Netmask.isIPv4` and :func:`Netmask.isIPv6` instead. In C++ API these methods have been removed.
```
Comment (Habbie) [inline docs/upgrading.rst]:
```suggestion
- Netmask class methods ``isIpv4`` and ``isIpv6`` have been deprecated in Lua, use :func:`Netmask.isIPv4` and :func:`Netmask.isIPv6` instead. In the internal C++ API these methods have been removed.
```

File: pdns/recursordist/docs/upgrade.rst [hunk 4]
```diff
@@ -4,9 +4,14 @@ Upgrade Guide
 Before upgrading, it is advised to read the :doc:`changelog/index`.
 When upgrading several versions, please read **all** notes applying to the upgrade.
 
-4.1.x to 4.2.0 or master
+4.2.x to 4.3.0 or master
 ------------------------
 
+- Netmask class methods ``isIpv4`` and ``isIpv6`` has been deprecated in Lua, use :func:`Netmask.isIPv4` and :func:`Netmask.isIPv6` instead. In C++ API these methods have been removed.
```
Comment (Habbie) [inline pdns/recursordist/docs/upgrade.rst]:
```suggestion
- Netmask class methods ``isIpv4`` and ``isIpv6`` have been deprecated in Lua, use :func:`Netmask.isIPv4` and :func:`Netmask.isIPv6` instead. In the internal C++ API these methods have been removed.
```

**Pull request merged**